cmd/nsdns/watch.go

```diff
@@ -0,0 +1,192 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+import (
+	"github.com/Eagerod/kube-namesilo-dns/pkg/icanhazip"
+	"github.com/Eagerod/kube-namesilo-dns/pkg/namesilo_api"
+	"github.com/Eagerod/kube-namesilo-dns/pkg/nsdns"
+)
+
+func watchCommand() *cobra.Command {
+	var ingressClass string
+	var domainName string
+
+	watchCmd := &cobra.Command{
+		Use:   "watch",
+		Short: "watch ingress objects, and create dns records",
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
This command relies on a lot of closures. Would be nice if the dependency structure was more clear.